vllm/spec_decode/spec_decode_worker.py

```diff
@@ -320,7 +348,7 @@ def init_device(self) -> None:
                 "[Speculative Decoding] Use MQA scorer for scoring proposals.")
 
         self.scorer = scorer_cls(scorer_worker=self.scorer_worker,
-                                 device=self.device,
+                                 device=self.device.type,
```

Comment:
The argument is `device` so you shouldn't pass "device type". You could take the device type in `scorer_cls` and don't need to change this line.

Reply:
Hi, @comaniac , the reason I changed that is because the device type is str in Scorer_cls init, but for some reason, it passed device=> so it failed mypy test

https://github.com/vllm-project/vllm/blob/main/vllm/spec_decode/interfaces.py#L78-L79